frontend/src/metabase/lib/formatting/date.tsx

```diff
@@ -55,6 +55,462 @@ const DATE_STYLE_TO_FORMAT: DATE_STYLE_TO_FORMAT_TYPE = {
+          output: "11:00–59 AM",
+          input: ["2018-01-01T11:00"],
+        },
+      },
+      {
+        same: null,
+        format: [TA, TA],
+        dashPad: " ",
+        test: {
+          output: "11:00 AM – 4:59 PM",
+          input: ["2018-01-01T11:00", "2018-01-01T16:00"],
+        },
+      },
+    ],
+    minute: [
+      {
+        same: "minute",
+        format: [MDYTA],
+        test: {
+          output: "January 1, 2018, 11:20 AM",
+          input: ["2018-01-01T11:20"],
+        },
+      },
+      {
+        same: "day",
+        format: [MDYTA, TA],
+        dashPad: " ",
+        test: {
+          output: "January 1, 2018, 11:20 AM – 2:35 PM",
+          input: ["2018-01-01T11:20", "2018-01-01T14:35"],
+        },
+      },
+      {
+        same: "year",
+        format: [MDTA, MDYTA],
+        dashPad: " ",
+        test: {
+          output: "January 1, 11:20 AM – February 2, 2018, 2:35 PM",
+          input: ["2018-01-01T11:20", "2018-02-02T14:35"],
+        },
+      },
+      {
+        same: null,
+        format: [MDYTA, MDYTA],
+        dashPad: " ",
+        test: {
+          output: "January 1, 2018, 11:20 AM – January 2, 2019, 2:35 PM",
+          input: ["2018-01-01T11:20", "2019-01-02T14:35"],
+        },
+      },
+    ],
+    "minute-of-hour": [
+      {
+        same: "minute",
+        format: [mmS],
+        test: {
+          output: "minute :05",
+          input: ["2018-01-01T11:05"],
+        },
+      },
+      {
+        same: null,
+        format: [mmP, "mm"],
+        test: {
+          output: "minutes :05–30",
+          input: ["2018-01-01T11:05", "2018-01-01T11:30"],
+        },
+      },
+    ],
+  };
+})();
+
+export const SPECIFIC_DATE_TIME_UNITS: DatetimeUnit[] = [
+  "year",
+  "quarter",
+  "quarter-of-year",
+  "month",
+  "month-of-year",
+  "week",
+  "week-of-year",
+  "day",
+  "day-of-week",
+  "day-of-month",
+  "day-of-year",
+  "hour",
+  "hour-of-day",
+  "minute",
+  "minute-of-hour",
+];
+
+export function genDateRangeTable() {
+  const getSpecsRows = (unit: DatetimeUnit) => {
+    const specs = DATE_RANGE_FORMAT_SPECS[unit];
+    return specs.map(({ same, test: { output } }) => {
+      const kind =
+        same === unit.split("-")[0]
+          ? "single"
+          : same
+          ? `same ${same}`
+          : "otherwise";
+      return [`<em>${kind}</em>`, output].map(s => `<td>${s}</td>`);
+    });
+  };
+  const getUnitRows = (unit: DatetimeUnit) => {
+    const rows = getSpecsRows(unit);
+    rows[0] = [
+      `<td valign=top rowspan="${rows.length}"><code>${unit}</code></td>`,
+      ...rows[0],
+    ];
+    return rows.map(cols => ["<tr>", ...cols, "</tr>"].join("\n"));
+  };
+  const headerRow = ["date unit", "range type", "example"]
+    .map(s => `<td><strong>${s}</strong></td>`)
+    .join("\n");
+  const allRows = SPECIFIC_DATE_TIME_UNITS.map(getUnitRows).flat();
+  return ["<table>", headerRow, ...allRows, "</table>"].join("\n");
+}
```

Comment:
Is this function used anywhere? Did you only use it to generate the table for this PR?

Reply:
It’s not used anywhere except to generate the PR description for review.

Reply:
Let's delete it then, I feel it may confuse people reading the code in the future if they see this function and can't understand what it's for or if it's important.

Reply:
672e368498759cd5f30d0d5c669158161549beda